packages/destination-actions/src/destinations/fullstory/identifyUserV2/index.ts

```diff
@@ -0,0 +1,71 @@
+import type { ActionDefinition } from '@segment/actions-core'
+import type { Settings } from '../generated-types'
+import { createUserRequestParams } from '../request-params'
+import { normalizePropertyNames } from '../vars'
+
+const action: ActionDefinition<Settings> = {
+  title: 'Identify User V2',
+  description: 'Sets user identity variables, creates user if it does not exist',
+  platform: 'cloud',
+  defaultSubscription: 'type = "identify"',
+  fields: {
+    uid: {
+      type: 'string',
+      required: true,
+      description: "The user's id",
+      label: 'User ID',
+      default: {
+        '@path': '$.userId'
+      }
+    },
+    display_name: {
+      type: 'string',
+      required: false,
+      description: "The user's display name",
+      label: 'Display Name',
+      default: {
+        '@path': '$.traits.name'
+      }
+    },
+    email: {
+      type: 'string',
+      required: false,
+      description: "The user's email",
+      label: 'Email',
+      default: {
+        '@path': '$.traits.email'
+      }
+    },
+    properties: {
+      type: 'object',
+      required: false,
+      description: 'The Segment traits to be forwarded to FullStory',
+      label: 'Traits',
+      default: {
+        '@path': '$.traits'
+      }
+    }
+  },
+  perform: (request, { payload, settings }) => {
+    const { properties, uid, email, display_name } = payload
+
+    const normalizedProperties = normalizePropertyNames(properties, { camelCase: true })
```

Comment:
Do we need to continue camel casing for backward compatibility?

Reply:
I don't know the history of camel casing. I don't believe it is required for V2.

Reply:
If we're case sensitive on the server (I'm not sure if we are), I could see removing camel casing being a barrier for existing v1 action users migrating to v2. We also camel case in the browser variant, so I wonder if removing camel casing would lead to an unexpected divergence in behavior between our browser destination and cloud mode destination. I think as a general rule we want to be changing property names as little as possible, but in this case maybe there are good reasons to continue camel casing in the v2 variant. /cc @patrick-fs

Reply:
> I think as a general rule we want to be changing property names as little as possible

Yes this is the goal, since we'd like to respect whatever naming conventions customers use for their custom data fields. @ndonze-fs I'm thinking that this original camel-case application is based off of V1 field name constraints where type suffixes needed to be a part of the field name. Otherwise, I'm not sure why we would force camel casing.

I'd feel a little sad about pulling forward an implementation that potentially changes our customer's data model (like: they're used to "not camel cased" but we force them to use camel case in FullStory).

Customers have to opt-in to the V2 destination. What if we made "we'll respect your naming conventions, but you might need to updates some FSTA queries" part of the opt-in agreement?

Reply:
@patrick-fs The data points I'm thinking about:

- Our Segment browser destination [camel cases identify and setUserProperties custom vars](https://github.com/cowpaths/segmentio-action-destinations/blob/main/packages/browser-destinations/src/destinations/fullstory/identifyUser/index.ts#L68). I don't have the background for why this is. Maybe it was a convenient way at the time to strip invalid characters (e.g. spaces)?
- Our Segment browser destination [does not camel case event custom vars](https://github.com/cowpaths/segmentio-action-destinations/blob/main/packages/browser-destinations/src/destinations/fullstory/trackEvent/index.ts#L33).
- Since we were camel casing custom user vars in our Segment browser destination, we took the same approach with our [v1 identifyUser cloud mode destination](https://github.com/cowpaths/segmentio-action-destinations/blob/main/packages/destination-actions/src/destinations/fullstory/identifyUser/index.ts#L61).
- Since we were not camel casing custom event vars in our Segment browser destination, we took the [same approach with our v1 trackEvent cloud mode destination](https://github.com/cowpaths/segmentio-action-destinations/blob/main/packages/destination-actions/src/destinations/fullstory/trackEvent/index.ts#L73) -- we didn't type suffix custom event vars.

That seems to raise questions around whether breaking parity between our v1 browser destination and v2 cloud mode destination will be problematic and whether dropping camel casing on the v2 cloud mode destination will have ramifications for existing customers. To your point, we could communicate any such ramifications to customers making the migration.

Reply:
> breaking parity between our v1 browser destination and v2 cloud mode destination will be problematic and whether dropping camel casing on the v2 cloud mode destination will have ramifications for existing customers

The implication is that the same set of user properties sent to browser destinations will be named differently when sent to server destinations, at least until there's a v2 browser destination.

I talked with Josh Rynne about not having a V2 browser destination available when the V2 cloud destination is available. He wasn't concerned at the time since he didn't think there was a lot of overlap between browser and cloud destination customers.

So, by that logic we might be all right. WDYT?

Reply:
If we're okay with breaking parity between v1 browser identify and v2 cloud mode identify -- and if we're comfortable with property names potentially changing between v1 cloud mode identify to v2 cloud mode identify (i.e. no longer being camel cased) -- I think we'd be all right. You mentioned us not being case sensitive on the server, so the casing alone wouldn't likely be an issue. If the camel casing lodash function was also changing other characters we could end up with legitimately different property names. We could dig into this more.

Reply:
Looking at the lodash documentation for [camelCase](https://lodash.com/docs/#camelCase) and our own transformation, `stripCharsNotSupportedInPropertyNames`, we would be losing removal of `-` characters.

Reply:
>  if we're comfortable with property names potentially changing between v1 cloud mode identify to v2 cloud mode identify (i.e. no longer being camel cased)

I'm comfortable with this, since I think this actually gets us more to where we need to be in respecting our customers' data models. From this perspective: v1 mishandled their property names, and v2 corrects this.

> `stripCharsNotSupportedInPropertyNames`, we would be losing removal of `-` characters

In this case I suppose there are reasons why we can't support hyphens related to indexing? It would be a good idea to dig into why there are characters we can't support that are reasonable for use in field names. For now, leave it in.

Reply:
> For now, leave it in.
For clarity, are you saying to leave in the `camelCase` transformation?  I could modify the `stripCharsNotSupportedInPropertyNames` to strip `-`

Reply:
Sorry that was ambiguous:

> are you saying to leave in the camelCase transformation?

No sir, take that out.

> I could modify the stripCharsNotSupportedInPropertyNames to strip -

I meant don't make any changes outside of removing camelCase. The way I read the thread was that `-` is currently being removed, so leave that piece alone. But, it would be good to understand why we're stripping out characters that are otherwise safe for JSON field names.

Reply:
I believe this is once again an attempt to keep names in line with the browser requirements as found [here](https://help.fullstory.com/hc/en-us/articles/360020623234#property-name-requirements).

Reply:
Cool, I was more getting to why we couldn't support things outside of alpha-numerics and `_`. No one on Eco can answer this question, I think the answer would come from the Indexing team.

In any case, we're good to move forward with removing the camel-case enforcement but keeping the character constraints.